cmd/libs/go2idl/go-to-protobuf/package.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+)
+
+func newProtobufPackage(packagePath, packageName string) *protobufPackage {
+	pkg := &protobufPackage{
+		PackageName: packageName,
+		PackagePath: packagePath,
```

Comment:
For the proto idl it felt right to have it in the same go package, but I'm
not locked into it.  It also means the proto import path is consistent with
the package name of the API (k8s.api.unversioned) vs having a different
import path than the package.

When we have the generated marshallers, those *have* to be in the same
package (since MarshalTo is a method on the struct).  The alternatives are
wrappers we add with a type switch just prior to serialization, but that is
somewhat yucky too.  But not there yet so can defer it till then.

On Nov 2, 2015, at 4:10 PM, Daniel Smith <notifications@github.com> wrote:

In cmd/libs/go2idl/go-to-protobuf/package.go
<https://github.com/kubernetes/kubernetes/pull/16649#discussion_r43682658>:

> +
> +import (
> +	"fmt"
> +	"os"
> +	"path/filepath"
> +	"sort"
> +	"strings"
> +
> +	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
> +	"k8s.io/kubernetes/cmd/libs/go2idl/types"
> +)
> +
> +func newProtobufPackage(packagePath, packageName string) *protobufPackage {
> +	pkg := &protobufPackage{
> +		PackageName: packageName,
> +		PackagePath: packagePath,

Maybe put generated files in a different directory? (.../generated,
.../proto, etc)

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16649/files#r43682658>.